src/ZEO/nagios.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import print_function
+##############################################################################
+#
+# Copyright (c) 2011 Zope Foundation and Contributors.
+# All Rights Reserved.
+#
+# This software is subject to the provisions of the Zope Public License,
+# Version 2.1 (ZPL).  A copy of the ZPL should accompany this distribution.
+# THIS SOFTWARE IS PROVIDED "AS IS" AND ANY AND ALL EXPRESS OR IMPLIED
+# WARRANTIES ARE DISCLAIMED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF TITLE, MERCHANTABILITY, AGAINST INFRINGEMENT, AND FITNESS
+# FOR A PARTICULAR PURPOSE.
+#
+##############################################################################
+"""%prog [options] address
+
+Where the address is an IPV6 address of the form: [addr]:port, an IPV4
+address of the form: addr:port, or the name of a unix-domain socket file.
+"""
+import json
+import optparse
+import os
+import re
+import socket
+import struct
+import sys
+import time
+
+NO_TRANSACTION = '0'*16
+
+nodiff_names = 'active_txns connections waiting'.split()
+diff_names = 'aborts commits conflicts conflicts_resolved loads stores'.split()
+
+per_times = dict(seconds=1.0, minutes=60.0, hours=3600.0, days=86400.0)
+
+def new_metric(metrics, storage_id, name, value):
+    if storage_id == '1':
+        label = name
+    else:
+        if ' ' in storage_id:
+            label = "'%s:%s'" % (storage_id, name)
+        else:
+            label = "%s:%s" % (storage_id, name)
+    metrics.append("%s=%s" % (label, value))
+
+def result(messages, metrics=(), status=None):
+    if metrics:
+        messages[0] += '|' + metrics[0]
+    if len(metrics) > 1:
+        messages.append('| ' + '\n '.join(metrics[1:]))
+    print('\n'.join(messages))
+    return status
+
+def error(message):
+    return result((message, ), (), 2)
+
+def warn(message):
+    return result((message, ), (), 1)
+
+def check(addr, output_metrics, status, per):
+    m = re.match(r'\[(\S+)\]:(\d+)$', addr)
+    if m:
+        addr = m.group(1), int(m.group(2))
+        s = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
+    else:
+        m = re.match(r'(\S+):(\d+)$', addr)
+        if m:
+            addr = m.group(1), int(m.group(2))
+            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        else:
+            s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    try:
+        s.connect(addr)
+    except socket.error:
```

Comment:
Why not ``except socket.error as err``?  Does ZEO support Python older than 2.6?